### PR TITLE
upgrade: deno_doc, deno_lint, dprint, swc

### LINKED ISSUE #8088: Bad import.meta handling in bundles
Input:
```ts
console.log(import.meta);
```

Actual output:
```js
console.log(import.meta);
```

Expected output:
```js
const import_meta_1 = { url: "file:///path/to/original/source", main: import.meta.main /* or false */ };
console.log(import_meta_1);
```

---

We're here again: https://github.com/swc-project/swc/issues/1115#issuecomment-700638446. `import.meta` is an object that can be wholly referenced and is even writable. SWC should not substitute individual fields, but call back to the loader for the initialising value of `import_meta_1`.

cc @kitsonk @kdy1

### LINKED ISSUE #8211: Uncaught ReferenceError when running bundle created with Deno 1.5.x
I'm getting runtime errors when executing bundles created with Deno `1.5.x` that I didn't get in `1.4.6`:

`main.ts`:
```
example % cat main.ts 
import * as luxon from 'https://unpkg.com/luxon@1.25.0/src/luxon.js';
```

Link to module for examination: https://unpkg.com/luxon@1.25.0/src/luxon.js

Bundling with `1.5.1`:
```
example % deno --version
deno 1.5.1
v8 8.7.220.3
typescript 4.0.3

example % deno bundle main.ts main-1.5.1.bundle.js
Bundle file:///Users/jesse/example/main.ts
Check file:///Users/jesse/example/main.ts
Emit "main-1.5.1.bundle.js" (102.86KB)

example % deno run main-1.5.1.bundle.js 
error: Uncaught ReferenceError: Cannot access 'Zone' before initialization
class IANAZone extends Zone {
                       ^
    at file:///Users/jesse/example/main-1.5.1.bundle.js:2015:24

example %
```

Bundling with `1.5.0`:
```
example % deno --version
deno 1.5.0
v8 8.7.220.3
typescript 4.0.3

example % deno bundle main.ts main-1.5.0.bundle.js
Bundle file:///Users/jesse/example/main.ts
Check file:///Users/jesse/example/main.ts
Emit "main-1.5.0.bundle.js" (102.88KB)

example % deno run main-1.5.0.bundle.js 
error: Uncaught ReferenceError: Cannot access 'Zone' before initialization
class IANAZone extends Zone {
                       ^
    at file:///Users/jesse/example/main-1.5.0.bundle.js:2015:24

example %
```

Bundling with `1.4.6`:
```
example % deno --version
deno 1.4.6
v8 8.7.220.3
typescript 4.0.3

example % deno bundle main.ts main-1.4.6.bundle.js
Bundle file:///Users/jesse/example/main.ts
Emit "main-1.4.6.bundle.js" (200KB)

example % deno run main-1.4.6.bundle.js 

example %
```

Running the `1.4.6` bundle with `1.5.1`:
```
example % deno --version               
deno 1.5.1
v8 8.7.220.3
typescript 4.0.3

example % deno run main-1.4.6.bundle.js 

example %
```

Any ideas?

### LINKED ISSUE #8224: Bundling has infinite loop with certain circular dependency scenarios
New bundler gets into an infinite loop and never completes the bundle.

This used to work on versions <1.5. Unfortunately can't share the code and I don't know how to minimize the testcase...

### LINKED ISSUE #8226: Linter shouldn't support eslint ignore directive
Currently `deno lint` supports two distinct ways of ignoring diagnostics on next line:
- `eslint-disable-next-line`
- `deno-lint-ignore`

In https://github.com/denoland/deno_lint/issues/476 @nayeemrmn suggests to make `deno-lint-ignore` take into account whole AST node which would render `eslint-disable-next-line` irrelevant. Additionally supporting eslint's directive leads to a subtle bug in `deno_lint` (https://github.com/denoland/deno_lint/issues/287). 

I believe we should remove support for `eslint-disable-next-line` and stay only with `deno-lint-ignore`.

### PR DESCRIPTION
**Pull request by bartlomieju**
Closes #8226
Closes #8211
Closes #8088
Closes #8224

### CODE REVIEW

File: cli/Cargo.toml [hunk 1]
```diff
@@ -64,8 +64,8 @@ serde = { version = "1.0.116", features = ["derive"] }
 sys-info = "0.7.0"
 sourcemap = "6.0.1"
 swc_bundler = "=0.14.1"
```
Comment (nayeemrmn) [inline cli/Cargo.toml:66]: Can we upgrade the bundler? It requires https://github.com/swc-project/swc/blob/8e07d7c7a6366801909565bf7e2b8d5762dd3edd/bundler/tests/deno.rs#L254-L289.

Comment (bartlomieju) [reply cli/Cargo.toml:66]: Yes, I will do that, but it requires some changes need to look into that

Comment (nayeemrmn) [reply cli/Cargo.toml:66]: Yes I upgraded the `import.meta` hook API in https://github.com/swc-project/swc/pull/1195, I linked the new required implementation above.

Comment (bartlomieju) [reply cli/Cargo.toml:66]: Okay, fixed, PTAL

**Review by nayeemrmn — APPROVED** [review]
Closes #8088.

LGTM

**Review by kitsonk — APPROVED** [review]
LGTM

**Pull request merged**